pkgs/stdenv/generic/setup.sh

```diff
@@ -182,31 +182,31 @@ _addRpathPrefix() {
 
 # Return success if the specified file is an ELF object.
 isELF() {
-    local fn=$1
+    local fn="$1"
     local fd
     local magic
     exec {fd}< "$fn"
-    read -r -n 4 -u $fd magic
+    read -r -n 4 -u "$fd" magic
```

Comment:
I don't even know what that `exec {fd}< "$fn"` means. Is `fd` guaranteed to have no spaces here?